### PR TITLE
update to analyzer 3.4 APIs

### PR DESCRIPTION
**Pull request by pq**
/cc @scheglov @bwilkerson

### CODE REVIEW

File: test/rule_test_support.dart [hunk 1]
```diff
@@ -388,7 +388,7 @@ abstract class _ContextResolutionTest with ResourceProviderMixin {
       throw StateError('Only dart files can be changed after analysis.');
     }
 
-    return super.newFile(path, content: content);
+    return super.newFile2(path, content);
```
Comment (scheglov) [inline test/rule_test_support.dart:391]: I would recommend to rename the containing method to `newFile2`, with corresponding making `content` a required positional parameter.

**Review by scheglov — APPROVED** [review]

**Comment by pq** [comment]
Sorry for the noise here.  I hadn't run the tests and am looking at the support I added for `unnecessary_getters_setters`.  Given that you can't have a non-final field in an enum, it seems like the lint should just not fire.

WDYT?

File: test/rule_test_support.dart (hunk 1 above)
Comment (pq) [reply test/rule_test_support.dart:391]: Done!

**Comment by bwilkerson** [comment]
> Given that you can't have a non-final field in an enum, it seems like the lint should just not fire.

I don't understand why not. If the user writes
```
enum E {
  a = E(1),
  b = E(2);
  final int _value;
  const E(this._value);
  int get value => _value;
}
```
Wouldn't it be valid to suggest changing the field's name to be that of the getter and removing the getter?

**Review by bwilkerson — APPROVED** [review]

**Pull request merged**